staging/src/k8s.io/apiserver/pkg/server/options/serving.go

```diff
@@ -219,33 +226,82 @@ func (s *SecureServingOptions) ApplyTo(config **server.SecureServingInfo) error
 	*config = &server.SecureServingInfo{
 		Listener:                     s.Listener,
 		HTTP2MaxStreamsPerConnection: s.HTTP2MaxStreamsPerConnection,
+		// We need an empty cert pool by default, so that the DelegatingAuthenticationOptions can possibly add certs to it
+		ClientCA: x509.NewCertPool(),
 	}
 	c := *config
 
-	serverCertFile, serverKeyFile := s.ServerCert.CertKey.CertFile, s.ServerCert.CertKey.KeyFile
-	// load main cert
-	if len(serverCertFile) != 0 || len(serverKeyFile) != 0 {
-		tlsCert, err := tls.LoadX509KeyPair(serverCertFile, serverKeyFile)
-		if err != nil {
-			return fmt.Errorf("unable to load server certificate: %v", err)
+	currentConfig := new(atomic.Value)
+
+	initialConfig, err := s.buildTLSConfig(c)
+	if err != nil {
+		return err
+	}
+
+	currentConfig.Store(initialConfig)
```

Comment:
if we build/store the initial config in Serve(), it can honor the SNICert change made by the loopback ApplyTo call